### ISSUE TITLE
Non-latin Keyboard Layouts

### ISSUE THREAD

**Issue by palasso** [solus-project/os-installer#28]
**Description**
Selecting a non-latin keyboard layout results in a dead-end in the installer.

**More Details**
The Keyboard Layout selection happens before setting hostname, username, password etc.
If the Keyboard Layout is selected to be a non-latin layout this will mean that some of the next steps won't be able to be completed. Specifically it won't be possible to set a hostname leading into a dead end.

**Workaround**
Go back on previous steps and select a latin layout. Finish the installation. Boot on the installed system and configure it to include a non-latin keyboard.

**When this happens**
This problem happens whenever the keyboard layout is selected to be a non-latin keyboard layout. There are two cases this happens:

1. The user chooses a non-latin keyboard manually
2. The installer suggests a non-latin keyboard to the user which the user most probably accepts (when finding the user's location automatically and the user lives in a country that normally uses a non-latin keyboard)

**Suggestions**
This issue could be resolved in various ways:
1. Moving the keyboard layout selection as the final option.
This isn't optimal as the user won't be able to enter their username and password afterwards when booting on the system as they won't have a latin keyboard layout.

2. Keeping the English keyboard layout and asking the user to select a secondary keyboard layout.
If the user tries to set hostname, username etc. with non-latin characters it might be helpful for the message to inform the user that specifically **latin** letters are required for those fields.
The current message for hostname is:
> Hostnames must be **lowercase**, and only contain *letters, numbers, hyphens and underscores*.Hostnames must also start with a lowercase letter.

3. Disabling all non-latin keyboard layouts from being selected on the installer. This would also mean the installer would never suggest a non-latin keyboard. The user would add a non-latin keyboard after the installation procedure on the installed system.

**UPDATE:** If suggestion 2 were to be implemented one might want to think which English layout shall be the default (there are many e.g. EN(US), EN(UK) etc.)?  The installer on first selection has a list of languages (those are the languages on which the installer will appear) and for English it says `English (United States)`.

**Comment by ghost** [comment]
I registered on github just to add my voice to this post. Pls fix it.

**Comment by amireldor** [comment]
Yup. Had to install English-only and only then configure Hebrew (my alternate language). After selecting Hebrew and being asked for hostname or username, I could only enter Hebrew letters and not switch to English.

I also think that once choosing your local layout, a tip about selecting the English layout again should be shown (win+space as default, for example).
